npm/src/s3upload/tdr-fetch-handler.ts

```diff
@@ -0,0 +1,126 @@
+import { HttpHandler, HttpRequest, HttpResponse } from "@aws-sdk/protocol-http"
+import { HttpHandlerOptions, HeaderBag } from "@aws-sdk/types"
+
+function requestTimeout(timeoutInMs = 0): Promise<never> {
+  return new Promise((resolve, reject) => {
+    if (timeoutInMs) {
+      setTimeout(() => {
+        const timeoutError = new Error(
+          `Request did not complete within ${timeoutInMs} ms`
+        )
+        timeoutError.name = "TimeoutError"
+        reject(timeoutError)
+      }, timeoutInMs)
+    }
+  })
+}
+
+export interface FetchHttpHandlerOptions {
+  /**
+   * The number of milliseconds a request can take before being automatically
+   * terminated.
+   */
+  requestTimeout?: number
```

Comment:
Maybe change the name of the param to this, just to make it a bit clearer is milliseconds: `requestTimeoutMs`

Reply:
Yeah makes sense, I've changed it here and where it's called.